contracts/consensus/Consensus.sol

```diff
@@ -598,10 +611,21 @@ contract Consensus is MasterCopyNonUpgradable, CoreLifetimeEnum, MosaicVersion,
 
         // Join in core contract.
         (uint256 validatorCount, uint256 minValidatorCount) =
-            CoreI(_core).joinDuringCreation(msg.sender);
+            CoreI(core).joinDuringCreation(msg.sender);
 
         if (validatorCount >= minValidatorCount) {
-            coreLifetimes[_core] = CoreLifetime.genesis;
+            coreLifetimes[core] = CoreLifetime.genesis;
+            ConsensusGatewayI consensusGateway = consensusGateways[_metachainId];
+            require(
+                address(consensusGateway) != address(0),
+                "Consensus gateway does not exist for given metachain."
```

Comment:
Can `consensusGateway` address be `0`? 
This should never happen.
Either remove this or make this assert.

Reply:
Updated with assert.